truffle/src/main/ruby/core/proc.rb

```diff
@@ -55,8 +55,8 @@ def curry(curried_arity = nil)
       raise ArgumentError, "cannot create binding from f proc"
     end
```

Comment:
Just tried removing that code, and there's a spec that tests binding of curried procs fails with an ArgumentError.